subprojects/platform-jvm/src/main/java/org/gradle/jvm/tasks/Jar.java

```diff
@@ -202,4 +218,25 @@ public CopySpec getMetaInf() {
     public CopySpec metaInf(Closure<?> configureClosure) {
         return ConfigureUtil.configure(configureClosure, getMetaInf());
     }
+
+    /**
+     * Provides a comparator that sorts the contents of a Jar/Ear/War so that META-INF is always first.
+     */
+    private static class JarContentsComparator implements Comparator<RelativePath> {
+        @Override
+        public int compare(RelativePath o1, RelativePath o2) {
+            boolean o1MetaInf = isMetaInf(o1.getSegments()[0]);
```

Comment:
Is there any requirement on the order?  The tests I see now that check for a particular order had something to do with the order of META-INF vs WEB-INF.